app/models/manageiq/providers/redhat/inventory/parser/infra_manager.rb

```diff
@@ -25,7 +25,7 @@ def ems_clusters
 
       ems_ref = ManageIQ::Providers::Redhat::InfraManager.make_ems_ref(cluster.href)
 
-      persister.ems_clusters.find_or_build(ems_ref).assign_attributes(
+      persister.ems_clusters.build(
```

Comment:
Using find_or_build().assign_attributes() wasn't updating the secondary_refs because [`build_secondary_indexes_for`](https://github.com/ManageIQ/inventory_refresh/blob/v0.1.z/lib/inventory_refresh/inventory_collection/data_storage.rb#L51) was only being called on the `build` and not on the `assign_attributes`